### PR TITLE
Fix/audiofocus

### PR DESCRIPTION
**Pull request by cyodrew**
## Description

Fixes a bug that does not correctly abandon audio focus after disconnect.

## Breakdown

- Audio Focus not correctly abandoned when restoring audio state
- `abandonAudioFocusRequest(focusRequest: AudioFocusRequest)` called for API level >= 26 and `abandonAudioFocus {}` called for API level < 26 similarly to `requestAudioFocus` in the `setAudioFocus()` method.
- Documentation states same listener/focus request that was used for `requestAudioFocus` should be used for `abandonAudioFocus`

## Validation

- Add new unit tests to validate changes

## Submission Checklist
 - [x] The source has been evaluated for semantic versioning changes and are reflected in `gradle.properties`
 - [X] The `CHANGELOG.md` reflects any **feature**, **bug fixes**, or **known issues** made in the source code

### CODE REVIEW

File: audioswitch/src/main/java/com/twilio/audioswitch/selection/AudioDeviceManager.kt [hunk 1]
```diff
@@ -12,16 +13,17 @@ import com.twilio.audioswitch.android.LogWrapper
 private const val TAG = "AudioDeviceManager"
 
 internal class AudioDeviceManager(
-    private val context: Context,
-    private val logger: LogWrapper,
-    private val audioManager: AudioManager,
-    private val build: BuildWrapper,
-    private val audioFocusRequest: AudioFocusRequestWrapper
+        private val context: Context,
```
Comment (Alton09) [inline audioswitch/src/main/java/com/twilio/audioswitch/selection/AudioDeviceManager.kt:16]: Please run `./gradlew spotlessApply` to fix formatting.

File: audioswitch/src/test/java/com/twilio/audioswitch/selection/AudioDeviceSelectorTest.kt [hunk 2]
```diff
@@ -298,6 +298,40 @@ class AudioDeviceSelectorTest {
         )
     }
 
+    @Test
+    fun `deactivate should abandon audio focus using pre Android O method if api version is 26`() {
```
Comment (Alton09) [inline audioswitch/src/test/java/com/twilio/audioswitch/selection/AudioDeviceSelectorTest.kt:302]: Great job testing the API boundary conditions 👍

File: audioswitch/src/main/java/com/twilio/audioswitch/selection/AudioDeviceManager.kt (hunk 1 above)
Comment (cyodrew) [reply audioswitch/src/main/java/com/twilio/audioswitch/selection/AudioDeviceManager.kt]: Done!

**Comment by Alton09** [comment]
@cyodrew Thanks again for all your help so far with this new library! We would like to send you some Twilio swag to show our appreciation. Please reach out to me via email at jaltonqualls@twilio.com for more details if you are interested.

**Comment by cyodrew** [comment]
Thanks @Alton09! The library has helped me so I wanted to give back!
